generators/client/templates/react/src/main/webapp/app/modules/login/logout.tsx.ejs

```diff
@@ -24,15 +24,20 @@ import { logout } from 'app/shared/reducers/authentication';
 export const Logout = () => {
   const logoutUrl = useAppSelector(state => state.authentication.logoutUrl);
   const idToken = useAppSelector(state => state.authentication.idToken);
+  const clientId = useAppSelector(state => state.authentication.clientId);
   const dispatch = useAppDispatch();
 
   useLayoutEffect(() => {
     dispatch(logout());
     if (logoutUrl) {
       // if Keycloak, logoutUrl has protocol/openid-connect in it
-      window.location.href = logoutUrl.includes('/protocol')
-        ? logoutUrl + '?redirect_uri=' + window.location.origin
-        : logoutUrl + '?id_token_hint=' + idToken + '&post_logout_redirect_uri=' + window.location.origin;
+      if (logoutUrl.includes('/protocol')) {
+        window.location.href = `${logoutUrl}?redirect_uri=${window.location.origin}`;
+      } else if(logoutUrl.includes('auth0.com')) {
+        window.location.href = `${logoutUrl}?client_id=${clientId}&returnTo=${window.location.origin}`
```

Comment:
In the current OIDC implementations, we don't have a client ID in any of the frontend code. Maybe it can be grabbed from the `idToken`? It's just a base64 encoded string.

Reply:
```suggestion
        window.location.href = `${logoutUrl}?client_id=${clientId}&returnTo=${window.location.origin}`;
```